### DISCUSSION TITLE
[Request] Bonus Claimblock Commands

### DISCUSSION THREAD

**Discussion by AntZaro** [GriefPrevention/GriefPrevention#51] [Ideas]
I have a suggestion for a few commands regarding bonus claimblocks. Currently, I am aware of two commands:
/adjustbonusclaimblocks (adjust balance of one player)
/adjustbonusclaimblocksall (adjust balance of all online players)

I have two requests. One is for two new commands:
/setbonusclaimblocks (sets balance of a player to the specified value)
/setbonusclaimblocksall (sets balance of all online players to the specified value)

Second request is an argument for the above commands:
Argument to set/adjust bonus claim block balances for all players (online and offline). 

I feel these commands would be useful, because they allow greater control over managing claimblocks on your server. In the event of a map reset, these commands would allow more options in how to deal with claimblocks. You could leave players' accrued balances, set their bonus to zero, or set the bonus to whatever value you want. Or even have it so all players get 500 claimblocks if someone donates a certain amount to the server. These are just some of the ideas I could think of that people could use these commands for. Hopefully at least one of my two suggestions is considered. Thanks for reading! 

### Spigot/Craftbukkit/Paper and GriefPrevention version:
Using latest version of Spigot and GP at the time of this post. 1.11 and 16.4 respectively.

**Comment by bigpresh** [comment]
Came to suggest the same thing, after dealing with some problems on a server where some players lost their bonus claim blocks (long story - not GP's fault).

I wrote a script to parse the correct amounts from the logs, and wanted to just generate a load of commands to set each player's bonus claim blocks to the correct amount (which would be a no-op if they already had the correct amount), but I can only add blocks, not set the amount - so I now need to do some more complicated arsing about to get their current bonus claim blocks and calculate any adjustment required.

**Comment by bigpresh** [comment]
I wonder if it should be additional commands to set a particular value, or just new syntax, e.g. `/adjustbonusclaimblocks =1000` say, to set to 1000, rather than to add 1000?
